resources/channel.sub.js

```diff
@@ -0,0 +1,1003 @@
+         * any, on the server side.
+         */
+        async disconnectReader() {
+            this._send("disconnectReader");
+        }
+
+        /**
+         * Disconnect this channel on the server side.
+         */
+        async delete() {
+            this._send("delete");
+        }
+    };
+    self.SendChannel = SendChannel;
+
+    const recvChannelsCreated = new Set();
+
+    class RecvChannel extends Channel {
+        type = "read";
+
+        constructor(uuid) {
+            if (recvChannelsCreated.has(uuid)) {
+                throw new Error(`Already created RecvChannel with id ${uuid}`);
+            }
+            super(uuid);
+            this.eventListeners.set("message", new Set());
+        }
+
+        async connect() {
+            if (this.hasConnection()) {
+                return;
+            }
+            await super.connect(event => this.readMessage(event.data));
+        }
+
+        readMessage(data) {
+            let msg = JSON.parse(data);
+            this._dispatch("message", msg);
+        }
+
+        /**
+         * Wait for the next message and return it (after passing it to
+         * existing handlers)
+         *
+         * @returns {Promise} - Promise that resolves to the message data.
+         */
+        nextMessage() {
+            return new Promise(resolve => {
+                let fn = ({data}) => {
+                    this.removeEventListener("message", fn);
+                    resolve(data);
+                };
+                this.addEventListener("message", fn);
+            });
+        }
+    }
+
+    /**
+     * Create a new channel pair
+     *
+     * @returns {[RecvChannel, SendChannel]}
+     */
+    self.channel = function() {
+        let uuid = createUuid();
+        let recvChannel = new RecvChannel(uuid);
+        let sendChannel = new SendChannel(uuid);
+        return [recvChannel, sendChannel];
+    };
+
+    /**
+     * Create an unconnected channel defined by a `uuid` in
+     * ``location.href`` for listening for `RemoteGlobal
+     * <#RemoteGlobal>`_ messages.
+     *
+     * @returns {RemoteGlobalCommandRecvChannel} - Disconnected channel
+     */
+    self.global_channel = function() {
+        let uuid = new URLSearchParams(location.search).get("uuid");
+        if (!uuid) {
+            throw new Error("URL must have a uuid parameter to use as a RemoteGlobal");
+        }
+        return new RemoteGlobalCommandRecvChannel(new RecvChannel(uuid));
+    };
+
+    /**
+     * Start listening for `RemoteGlobal <#RemoteGlobal>`_ messages on
+     * a channel defined by a `uuid` in `location.href`
+     *
+     * @returns {RemoteGlobalCommandRecvChannel} - Connected channel
+     */
+    self.start_global_channel = async function() {
+        let channel = self.global_channel();
+        await channel.connect();
+        return channel;
+    };
+
+    self.closeAllChannelSockets = async function() {
```

Comment:
I would say the closest thing to consistent would be to rename this to `close_all_channel_sockets`, since the other global methods added by this file follow that pattern.